---
title: Environmentally robust *cis*-regulatory changes underlie rapid climatic adaptation

# Use letters for affiliations, numbers to show equal authorship and to indicate the corresponding author
author:
  - name: Mallory A. Ballinger
    affiliation: a,b,1,2
  - name: Katya L. Mack
    affiliation: c,1
  - name: Sylvia M. Durkin
    affiliation: a,b
  - name: Eric A. Riddell
    affiliation: a,d
  - name: Michael W. Nachman
    affiliation: a,b,2
address:
  - code: a
    address: Museum of Vertebrate Zoology, University of California, Berkeley, Berkeley, CA, 94720
  - code: b
    address: Department of Integrative Biology, University of California, Berkeley, Berkeley, CA, 94720
  - code: c
    address: Department of Biology, Stanford University, Stanford, CA, 94305
  - code: d
    address: Department of Ecology, Evolution, and Organismal Biology, Iowa State University, Ames, Iowa, 50011

corresponding_author:
  code: 2
  text: "To whom correspondence should be addressed. M.A.B., mallory.ballinger@berkeley.edu; M.W.N., mnachman@berkeley.edu"

# For footer text
lead_author_surname: Ballinger, Mack

equal_authors:
  code: 1
  text: "M.A.B. and K.L.M. contributed equally to this work."

author_contributions: |
  M.A.B. and M.W.N. designed research. M.A.B., S.M.D., and E.A.R. performed research. M.A.B. and K.L.M. analyzed data. M.A.B., K.L.M., and M.W.N. wrote the paper.

conflict_of_interest: |
  The authors declare no conflict of interest.

abstract: |
  Changes in gene expression have been proposed to play a major role in adaptive evolution. However, gene expression is highly context-dependent and very few studies have determined the influence of genetic and non-genetic effects on adaptive gene regulation in natural populations. Here, we utilize context-dependent allele-specific expression to characterize \textit{cis} and \textit{trans} changes underlying divergence in temperate and tropical house mice in two metabolic tissues under two thermal conditions. First, we show that gene expression divergence is pervasive between populations and across thermal conditions, with roughly 5-10% of genes exhibiting genotype-by-environment interactions. Second, we found that most intraspecific regulatory divergence was due to \textit{cis}-regulatory changes that were stable across temperatures. In contrast, patterns of expression plasticity were largely attributable to \textit{trans}-effects, which showed greater sensitivity to temperature. Nonetheless, we discovered a small subset of temperature-dependent \textit{cis}-regulatory changes, thereby identifying loci underlying expression plasticity. Finally, we performed scans for selection in wild house mice to identify genomic signatures of rapid adaptation. Genomic outliers were enriched in genes with evidence for \textit{cis}-regulatory divergence. Strikingly, these genes were associated with phenotypes that affected body weight and metabolism, identifying \textit{cis}-regulatory changes as a mechanism for adaptive body size evolution between populations. Together, these results support the central role of \textit{cis}-regulatory divergence in adaptive evolution over extremely short timescales.

significance: |
  Little is known about the relative contributions of genetic versus environmental factors to gene expression variation in natural populations. Here, we discovered that genetic effects were far more pervasive than environmental effects on gene expression differences between house mice that have rapidly adapted to new environments. Notably, many of these genetic effects are under selection and underlie adaptive body size differences between populations. However, we also identified a few genes controlling plastic expression across different environments. Our study is one of the few to identify loci associated with plasticity in natural populations and supports the central role of genetic effects in the form of \textit{cis}-regulatory divergence in adaptive evolution over extremely short timescales (a few hundred generations).

acknowledgements: |
  We thank Beth Dumont for providing whole genome sequences of MANA and SARA. We thank Yocelyn Guti&#233;rrez-Guerrero for providing exome sequences from a population from Manaus, Brazil. Funding and support for this work was provided by the National Institutes of Health (NIH; R01 GM074245 and R01 GM127468 to M.W.N.). This work used the Extreme Science and Engineering Discovery Environment (XSEDE), which is supported by National Science Foundation grant number ACI-1548562. M.A.B. was supported by a National Science Foundation Graduate Research Fellowship (DGE-1106400), a Junea W. Kelly Museum of Vertebrate Zoology Graduate Fellowship, and a University of California Berkeley Philomathia Graduate Fellowship. K.L.M. was supported by a Ruth Kirschstein National Research Service Award from NIH and a Stanford Center for Computational, Evolutionary and Human Genomics postdoctoral fellowship.

keywords:
  - adaptation
  - \textit{cis}-regulatory evolution
  - \textit{plasticity}-eQTL
  - \textit{Mus}

pnas_type: pnasresearcharticle

bibliography: pnas-sample.bib
csl: pnas.csl

lineno: true

output: rticles::pnas_article
---

A central goal in evolutionary biology is to understand how organisms adapt to novel environments. Gene regulation has long been recognized to play a major role in adaptive evolution [@King1975; @Wray2003], especially across short evolutionary timescales [e.g., refs. @Jones2012; @Fraser2013]. Yet, we still have a poor understanding of how changes in regulatory architecture shape adaptive evolution. *Cis*-regulatory elements (e.g. promoters, enhancers) are predicted to be the primary substrate of adaptive evolution as they tend to be less pleiotropic than protein-coding changes [@Wray2007; @Wittkopp2011; @Carroll2008; @Stern2008]. However, selection may favor divergence through *trans*-acting mechanisms (e.g., transcription factors), particularly when *trans*-effects modulate gene regulatory networks that are beneficial in new environments [@Cooper2003; @Stern2009]. *Trans*-effects may also play a significant role in plastic changes in gene expression [@Smith2008; @Tirosh2009; @Promislow2005], and selection on genetic variation underlying plasticity may facilitate adaptation to new environments [@West-Eberhard2003; @West-Eberhard2005]. However, determining the relative importance of *cis*- and *trans*-changes to adaptation is challenging given that gene regulation is highly dependent on the environment, tissue-type, sex, and developmental stage [@St_Pierre2022; @Lopes-Ramos2020; @He2021; @Hadadi2022; @Tung2015; @Naranjo2015]. In most studies, regulatory patterns are often quantified under a single context, limiting our understanding of how gene regulatory architecture shapes adaptive evolution in natural populations. \bigskip

The recent expansion of house mice into the Americas provides an opportunity to address the role of gene regulatory changes in adaptive evolution. Since their arrival from Western Europe ~500 years ago, house mice (*Mus musculus domesticus*) have rapidly adapted to various climatic extremes through changes in morphology, physiology, and behavior [@Lynch1992; @Ferris2021; @Phifer-Rixey2018; @Bittner2021; @Ballinger2022]. One striking example of this is changes in body size, as mice from more northern populations are significantly larger than mice closer to the equator, likely reflecting adaptation to thermal environments [@Ballinger2022]. Previous studies point to an important role for gene regulation in driving this local adaptation. First, genomic scans have primarily identified positive selection on noncoding regions [@Ferris2021; @Phifer-Rixey2018], which have been linked to differences in gene expression [@Phifer-Rixey2018; @Mack2018]. Second, changes in *cis*-regulation at specific loci have been associated with variation in body weight in North American mice [@Mack2018]. Finally, gene expression plasticity has been shown to differ between populations in response to environmental stressors [@Bittner2021], suggesting a role for context-specific regulatory divergence in local adaptation.

\begin{figure*}[t]
  \includegraphics[width=\textwidth]{./figure_1.pdf}
  \caption{\textbf{Evolved differences in phenotypes and gene expression. (A)} Variation in mean annual temperature across North and South America. Wild-caught individuals were collected in upstate New York (43$^{\circ}$N) and equatorial Brazil (3$^{\circ}$S). \textbf{(B)} Genetic differences in body mass (g), pelage conductance (W\textsuperscript{-1}m\textsuperscript{-2}C\textsuperscript{-1}), tail length (mm), and ear length (mm) between New York and Brazil. Tail length and ear length are plotted relative to body mass for each individual. Individuals are represented as individual points, and boxplots indicate the 25th, median, and 75th quartiles. Results from linear mixed models are presented in upper right corners (\textit{*P} < 0.05; Table S1). Males and females show similar patterns and are combined for plotting simplicity. \textbf{(C)} Common garden experimental design. Individuals were reared under two temperatures from weaning until adults. \textbf{(D)} Principal component plots for PC1 vs PC3 based on male gene expression in BAT and liver. PC1 separates individuals based on genotype while PC3 reflects environmental differences. \textbf{(E)} Expression divergence between New York and Brazil males in warm and cold for both BAT and liver. Log2 fold changes between parents were calculated for all genes independently. In each panel, points (representing individual genes) are colored depending on their direction and significance of the log2 fold change. Insets depict the total number of differentially expressed genes for each comparison (FDR < 0.05). Females show similar patterns and are depicted in Figures S2-S3.}
\end{figure*}
\vspace{-1pt}
Here, we investigate the role of gene regulation in adaptation in house mice from contrasting thermal environments. Specifically, using RNA-seq data collected from liver and brown adipose tissue in males and females, we measured gene expression in temperate and tropical mice and in their F1 hybrids when reared under warm and cold temperatures. This allowed us to describe the proportion of divergently expressed genes that are due to changes in *cis*, *trans*, or both, and to determine the degree to which *cis*- and *trans*-regulation is context-dependent. Finally, we performed scans for selection in wild populations of house mice to identify genomic signatures of adaptation. We then intersect these genomic outliers with genes exhibiting *cis*-regulatory divergence to identify putatively adaptive *cis*-regulatory mutations associated with local adaptation. Our results provide insight into how gene expression is regulated across multiple contexts and how this complex regulatory divergence within species may contribute to adaptive evolution.


Results {.unnumbered}
========================================

Extensive gene expression divergence between temperate and tropical house mice {.unnumbered}
-------------------
To characterize the regulatory architecture of adaptation, we first examined gene expression differences in mice from two drastically different environments in the Americas: Saratoga Springs, New York, USA (SARA), located at 43$^{\circ}$N, and Manaus, Amazonas, Brazil (MANA), located near the equator at 3$^{\circ}$S. Saratoga Springs and Manaus differ considerably in climate, such as mean annual temperature (Figure 1A), and mice from these environments show several phenotypic differences consistent with climatic adaptation. Specifically, mice from New York are larger, retain more heat through their fur, and have shortened extremities compared to mice from Brazil (ANOVA tests, *P* < 0.05) (Figure 1B; Table S1), suggesting adaptation to cold environments [@Ballinger2022].

\vspace{-1pt}
We explored patterns of gene expression evolution by rearing New York and Brazil mice under two temperatures (5$^{\circ}$C and 21$^{\circ}$C) and sequenced brown adipose tissue (BAT) and liver transcriptomes of 48 individuals (6 / line / sex / environment) (Figure 1C). We chose these two tissues as they play important roles in both metabolism and adaptive thermogenesis [@Abumrad2017; @Simcox2017; @Cannon2004]. Principal component analysis (PCA) of all gene expression data revealed tissue type as the largest source of variance (PC1 ~97% of variance explained), followed by sex (PC2 ~1.5%) (Figure S1). Within each tissue and sex, New York and Brazil mice cleanly separated along PC1 (>60% of variance explained), while PC3 largely separated warm- and cold-reared mice (>4% of variance explained) (Figures 1D, S2). We also identified more than a third of genes to be differentially expressed between New York and Brazil mice (false discovery rate (FDR) < 0.05) (Figures 1E, S3-S4), with most expression differences concordant across environments and sexes.

This strong pattern of divergence was also apparent when we categorized differentially expressed genes as those showing genetic variation (G), environmental variation [i.e., plasticity (E)], or genetic variation for plasticity (i.e., GxE) (Figures 2A, S3). Genotype had >1.5x larger effect size (calculated as the mean absolute value of the log2 fold change) on gene expression than environment across both tissues (Figures 2A, S3). Similar effects were identified when we attributed expression differences to genotype and sex, though these patterns were largely tissue-dependent (Figure S4). Overall, these results demonstrate that within sexes and tissues, genotype plays a larger role than either environment or GxE interactions in shaping expression differences between temperate and tropical house mice.

Reduced gene expression plasticity in cold-adapted mice {.unnumbered}
-------------------
Given that New York and Brazil mice have evolved under different thermal environments, we reasoned that gene expression responses to temperature would differ between these lines. Roughly ~5% and ~10% of all expressed genes showed significant GxE in liver and BAT, respectively (FDR < 0.05) (Figures 2B, S3). Notably, we found fewer differentially expressed genes in New York mice (~5% BAT; ~1% liver) than Brazil mice (~10% BAT; ~5% liver) (Chi-square tests, liver and BAT: *P* < 0.05), suggesting that New York mice may be more buffered against cold stress.

Next, we explored the relationship between plastic gene expression changes and evolved gene expression differences. Adaptive plasticity may facilitate the colonization of new environments by moving a population closer to the phenotypic optimum, while non-adaptive plasticity may do the opposite [@Price2003; @Ghalambor2007]. To determine if the pronounced temperature response of Brazil mice is adaptive or non-adaptive, we asked whether the direction of expression plasticity of Brazil mice correlates with expression divergence between New York and Brazil mice (see Methods). We found that expression plasticity generally goes in the same direction as evolved divergence for both tissues (positive Spearman’s correlations, *P* < 0.05) (Figures 2C, S3), consistent with patterns of adaptive plasticity [@Fischer2016; @Ghalambor2015; @Bittner2021]. These results suggest that plasticity may have facilitated the rapid expansion of house mice into new environments.

\begin{figure}[!t]
  \centering
  \includegraphics[width=1\columnwidth]{./figure_2.pdf}
  \caption{\textbf{Patterns of genotype-by-environment interactions (GxE). (A)} Ternary plots depicting the proportion of each gene’s expression variance explained by genotype (G), environment (E), and GxE. The relative proportion of each factor is shown for all differentially expressed male genes in BAT and liver. Total variance is the sum of all three components. \textbf{(B)} Comparison of gene expression differences between temperature regimes in NY and BZ males in both BAT and liver. Log2 fold changes between temperatures were calculated for all genes independently. In each panel, points (representing individual genes) are colored depending on their direction and significance of the log2 fold change. GxE categories include line-specific responses or opposite responses between lines. Insets depict the total number of differentially expressed genes for each comparison (FDR < 0.05). \textbf{(C)} The relationship between gene expression plasticity and evolved divergence in BAT and liver. Points represent expression differences with statistically significant plasticity in BZ (cold vs warm; FDR < 0.05) as well as significant expression divergence between NY and BZ at warm temperature (FDR < 0.05). Points colored in orange represent genes with a positive correlation between plasticity and evolved divergence and represent adaptive plasticity. Points in black represent genes with a negative association and represent non-adaptive plasticity. Insets depict the observed correlation coefficient (orange solid lines) is more positive than a randomized distribution of correlation coefficients for each tissue (see Methods for details). Asterisks denote significance of adaptive plasticity for each tissue (binomial exact tests, \textit{P} < 0.05). Females show similar patterns and are depicted in Figures S2-S3.}
\end{figure}

Expression divergence is predominantly due to *cis*-regulatory changes, and most *cis*-changes are robust to environmental temperature {.unnumbered}
-------------------
To investigate the gene regulatory mechanisms underlying expression differences between New York and Brazil mice, we generated BAT and liver RNA-seq from NY x BZ F1 hybrids reared in both warm and cold environments (Figures 1C, S5). Measuring gene expression in F1 hybrids allowed us to discern if parental gene expression differences are due to *cis*- and/or *trans*-acting changes by assessing patterns of allele-specific expression (ASE) (Figure 3A). Specifically, as F1 hybrids inherit both a Brazil allele and New York allele within in the same *trans*-acting environment, differences in expression between alleles are indicative of one or more *cis*-acting elements [@Cowles2002; @Wittkopp2004; @McManus2010]. In contrast, if no ASE is detected in hybrids but differences are observed between parental lines, we can infer divergence is likely due to *trans*-acting factors [@Cowles2002; @Wittkopp2004; @McManus2010].

\begin{figure*}[t]
  \includegraphics[width=\textwidth]{./figure_3.pdf}
  \caption{\textbf{The relative distribution of regulatory changes between New York and Brazil house mice across environments and tissues. (A)} Schematic depicting how \textit{cis}- and \textit{trans}-changes can be inferred with F1 hybrids, and how environmental differences may result in \textit{cis} x temperature and \textit{trans} x temperature effects. Blue and gold boxes represent \textit{cis}-regulatory regions for NY and BZ, respectively. Wavy lines depict transcript levels of an allele. TF = transcription factor. \textbf{(B)} Points (individual genes) represent log2 fold changes between reads mapping to each allele in the hybrid (BZ allele / NY allele; y-axis) and the reads mapping to each parental line (BZ parent / NY parent; x-axis). Genes are colored based on their inferred regulatory category: orange = $\textit{cis}$, purple = $\textit{trans}$, green = $\textit{cis}  \& \textit{trans}$, gray = conserved or ambiguous. Genes categorized as conserved or ambiguous (gray points) constitute roughly $75\%$ of all genes and are centered on the origin and mostly hidden behind other genes. \textbf{(C)} Changes in the number of genes for each inferred regulatory category between temperature regimes are illustrated in the alluvial plot. Genes that were conserved or ambiguous (gray) at both temperatures are not shown.}
\end{figure*}

We tested 5,898 genes for ASE based on the presence of fixed differences between parental Brazil and New York lines (see Methods). While most genes showed conserved gene regulation between New York and Brazil mice (~75%), genes with evidence for expression divergence tended to involve changes in *cis* (Figure 3B). Specifically, 7-8% of genes showed expression divergence due to *cis* alone and 5-6% genes showed evidence of divergence due to *cis* and *trans* (Figure 3B). Only ~5% of genes involved regulatory changes solely in *trans* (Figure 3B). Moreover, the magnitude of *cis*-effects were greater than *trans*-effects per gene (Wilcoxon signed-rank test, *P* < 2.2 x 10^-16^). The predominance of *cis*-regulatory changes relative to *trans*-changes is consistent with previous studies in house mice [@Goncalves2012; @Shen2014; @Mack2016].

We next asked how the environment modulates gene regulatory evolution by comparing patterns of *cis*- and *trans*-regulatory differences across environments. Similar to expression patterns observed in the parents, the majority of genes that could be categorized across temperature treatments showed the same regulatory control in both environments (~88%) (Figure 3C). For the genes that did show a change in regulatory control, we found that *cis*-regulatory changes were more insensitive to temperature than *trans*-changes. Comparing the difference in magnitude of the *cis*- and *trans*-differences between warm and cold conditions, we found that *trans*-differences were greater between environments for both tissues (Wilcoxon signed-rank tests, *P* < 2.2 x 10^-16^) (Figure S6). The cold environment also had a lower proportion of genes with *trans*-divergence (Chi-square tests; BAT, *P*=0.0003; liver, *P*=0.02), where the proportion of genes with only *cis*-divergence was the same across temperature conditions (Chi-square tests; BAT, *P*=0.51; liver, *P*=0.66). These results suggest that *trans*-effects play a larger role in gene expression plasticity than *cis*-effects.

A small number of genes show temperature-dependent *cis*-regulation {.unnumbered}
-------------------
While most *cis*-effects were robust to temperature, we were specifically interested in exploring whether any genes showed temperature-dependent *cis*-effects. Such genes are of particular interest since they correspond to *plasticity*-eQTL (i.e., loci that harbor mutations underlying a plastic response)[@Li2006]. To identify genes for which there was a significant effect of temperature on regulatory divergence, we determined if either the *cis* and/or the *trans* component showed a significant interaction with temperature (see Methods). We identified *cis* x temperature effects for 11 genes in BAT (*gstt1*, *wars2*, *hsd11b1*, *itih5*, *dst*, *tmed2*, *plbd1*, *cdh13*, *scd1*, *tmem45b*, *s100a13*) and 4 in the liver (*elovl3*, *hmgcs2*, *wars2*, *ebpl*) (FDR < 0.1). Most of these genes showed differences in the magnitude of ASE between temperatures, but we also observed cases where ASE was induced by one temperature treatment (i.e., *wars2*, *tmed2*, *cdh13*, *s100a13*, *ebpl*, *hmgcs2*). Over half of the genes corresponding to *plasticity*-eQTL showed a smaller plastic response in New York than in Brazil, consistent with the overall reduction in expression plasticity in cold-adapted mice. We also identified a small number of genes with significant *trans* x temperature effects in BAT (18 genes) and liver (1 gene) (FDR < 0.1) (Table S2). Several of these genes with temperature-induced regulatory differences have suggested roles in energy metabolism and thermal tolerance [e.g., refs @Fumagalli2015; @Hallmark2019; @Shore2013; @Westerberg2006]. The identification of temperature-dependent gene regulatory effects (especially *plasticity*-eQTL) indicates a role for evolved changes in plasticity between temperate and tropical mice.

\textit{Cis}-regulatory changes are largely tissue-specific and are enriched for body size and metabolism {.unnumbered}
----------------------
While both liver and BAT play essential roles in metabolism and thermogenesis, these tissues have distinct functional properties that differentiate their role in environmental adaptation. In both tissues, genes with evidence for *cis*-divergence were enriched for GO terms related to metabolic processes, as well as the pathway for metabolism (Reactome R-MMU-1430728; liver, FDR=6.55 x 10^-8^; BAT, FDR=1.49 x 10^-8^). Genes with *cis*-regulatory changes in the liver were enriched for several mutant phenotype annotations for homeostasis and metabolism, including abnormal lipid homeostasis (FDR=6.248 x 10^-5^), abnormal cholesterol level (FDR=0.003), abnormal energy expenditure (FDR=0.001), and abnormal triglyceride level (FDR=0.008). Additionally, genes with *cis*-changes in the liver showed a greater than 2-fold enrichment of genes with mutant phenotypes for abnormal susceptibility to weight gain (FDR=0.014) and were nominally significantly enriched for several other phenotypes related to body weight, size, and composition (Figure S7). Interestingly, two genes (*bcat2*, *adam17*) exhibiting *cis*-regulatory divergence were previously implicated in body weight differences in North American populations [@Mack2018], further supporting their role in adaptive divergence between house mouse populations.

Next, we assessed the extent to which regulatory control is tissue-biased. Comparing gene expression evolution in BAT and liver, we found regulatory divergence to be largely tissue-biased. The majority of genes (80%) for which we could assign a regulatory category in each tissue were assigned to a different regulatory category in the other tissue (2954/3672 genes). In particular, we found that *trans*-divergence was more likely to be restricted to one tissue (with expression conserved between lines in the other tissue), compared to *cis*-changes which were more often shared (>2-fold more) (Chi-square test *P* < 0.0001). This may reflect the general observation of increased tissue-specificity of *trans*-effects relative to *cis*-effects [@GTEx_Consortium2017].

To formally identify tissue-biased ASE, we contrasted ASE measurements in BAT and liver for paired hybrid samples (see Methods). We identified 338 genes with evidence for differential allele-specific expression between tissues (Figure S8). While the majority of these genes (77%) showed significant allele-specific expression in just one tissue, we also identified cases where allele-specific expression was present in both tissues but with differences in expression magnitude or direction (23%). Of these genes, forty-three had discordant allele-specific expression between tissues, where the opposite parental allele was up-regulated between tissues. Genes with tissue-biased ASE were enriched for metabolic phenotypes (e.g., abnormal lipid homeostasis, FDR=0.00027; increased food intake, FDR=0.036) and tissue specific functions and physiology (e.g., abnormal adipose tissue physiology, FDR=0.007; abnormal liver morphology, FDR=0.00097). These results highlight the importance of tissue-specific gene regulation in population divergence.

Positive selection on genes with *cis*-regulatory divergence in wild house mouse populations {.unnumbered}
------
As *cis*-regulatory variants are often drivers of local adaptation [@Fraser2010; @Fraser2011; @Fraser2013], and because most regulatory divergence between New York and Brazil house mice is governed in *cis*, we next explored whether genes regulated in *cis* are under positive selection in wild mice from the Americas. To test this, we utilized previously published whole exome data from wild-caught individuals collected from New Hampshire/Vermont, USA (NH/VT) [@Phifer-Rixey2018] and Manaus, Brazil (MAN) (Guti&#233;rrez-Guerrero et al., *in prep*), and compared these data to previously published whole genome data from Eurasian populations of house mice [@Harr2016]. Genetic PCA distinguished mice based on subspecies and population-of-origin (Figures 4A, S9), with mice from NH/VT clustering most closely with mice from Germany. These results are consistent with the suggestion that mice from eastern North America are most closely related to populations in northern Europe [@Tichy1994; @Morgan2022].

\begin{figure*}[t]
  \includegraphics[width=\textwidth]{./figure_4.pdf}
  \caption{\textbf{Genomic outliers are enriched in genes with evidence for} \textbf{\textit{cis}-regulatory divergence. (A)} Genetic PCA of wild house mice distinguished mouse populations based on population-of-origin (\textit{Mus musculus domesticus} (M.m.d.)) and subspecies (\textit{Mus musculus castaneus} (M.m.c.), \textit{Mus musculus musculus} (M.m.m.)). The x and y axes show the first and second SNP eigenvectors, respectively (EV; PC1: 29\% of variance, PC2: 8\% of variance. \textbf{(B)} Autosomal selection scan showing \textit{PBSn1} results for the New Hampshire/Vermont (NH/VT) focal population. Orange points depict genes that exhibit \textit{cis}-regulatory divergence and overlap with outlier regions. \textbf{(C)} Gene set enrichment analysis for genes with ASE that overlap genomic outliers in the NH/VT population. ASE outliers were highly enriched for mouse phenotypes related to body size differences and metabolic features, across both temperature treatments. \textbf{(D)} Candidate gene that exhibits \textit{cis}-regulatory divergence and overlaps with outlier region. Allele frequencies (pie charts) of significant SNPs (gold asterisks) in the four populations.}
\end{figure*}

Next, to identify genetic signatures of adaptation in house mice from the Americas, we performed a scan for regions of genetic differentiation consistent with selection using a normalized version of the population branch statistic (*PBSn1*). We used this test to identify highly differentiated loci in our focal populations in the Americas (MAN and NH/VT) relative to Eurasian populations (see Methods). In total, 83,538 and 84,420 non-overlapping 5-SNP windows were analyzed for Manaus and NH/VT, respectively. Outlier windows in NH/VT and MAN overlapped 538 and 530 genes, respectively (File S1).

Finally, we asked to what extent genomic divergence among wild mice from temperate and tropical environments is associated with *cis*-regulatory changes. Specifically, if natural selection associated with climatic adaptation has acted mainly on regulatory variants, we predicted an enrichment of *PBSn1* outliers near genes displaying ASE [e.g., ref. @York2018]. To test this prediction, we overlapped candidate regions for selection based on *PBSn1* outlier windows with genes for which we identified evidence for allele-specific expression in BAT or liver. In NH/VT, we found outlier windows overlapped 71 and 62 genes with evidence for *cis*-regulatory divergence under warm and cold conditions, respectively (overlap 44 genes) (Figure 4B; File S1). The overlap between genes with *cis*-regulatory divergence and outlier windows in this population was greater than expected by chance (hypergeometric test, *P*=0.0016) and genes with allele-specific expression were associated with higher average *PBSn1* scores than background genes (*P*=0.00026, see Methods). ASE outliers were enriched for mutant phenotypes related to body size, growth, and metabolism relative to other genes with *cis*-regulatory divergence (e.g., abnormal postnatal growth/weight/body size, abnormal susceptibility to weight gain, decreased susceptibility to diet-induced obesity, and increased energy expenditure; FDR < 0.05) (Figure 4C; File S1). This gene set also includes genes whose expression in the liver was previously associated with body mass variation in natural populations of North American house mice (*bcat2*, *col6a1*, *col5a2*, *col3a1*) [@Phifer-Rixey2018; @Mack2018]. Additionally, this set included genes implicated in obesity and metabolic phenotypes in humans (e.g., *wrn*, *plaat3*, *prkar2b*, *sulf2*, *smoc1*) (Figure 4D) [@Berube2013] and mice (Table S3). Together, these results suggest that selection has acted on *cis*-regulatory genes related to metabolism and body weight in New York mice.

In contrast, we did not find significant overlap between genes with allele-specific expression and *PBSn1* outliers for Manaus (*P*=0.4). Outlier windows overlapped 49 and 51 genes with evidence for *cis*-regulatory divergence under warm and cold conditions, respectively (Figure S10). Genes were not enriched for metabolic process terms or phenotypes. The significant overlap between *PBSn1* outliers and ASE in the temperate mice but not in the tropical mice suggests that adaptive gene expression differences may predominantly reflect adaptation to cold environments (rather than to warm environments).

Discussion {.unnumbered}
========================================

Understanding how both genetic and non-genetic factors influence gene expression is essential to understanding adaptive evolution. Here, we utilized allele-specific expression in liver and brown adipose tissue to characterize *cis* and *trans* changes underlying expression differences between temperate and tropical house mice when reared under warm and cold laboratory environments. We found that most regulatory divergence was governed by *cis*-regulatory variation, and that these *cis*-effects were largely independent of environmental temperature. However, a subset of genes showed temperature-dependent *cis*-effects and thus represent QTL for expression plasticity. We also found that many *cis*-regulated genes were associated with metabolism and body size, but that these *cis*-effects were often tissue-biased. Finally, overlap of genes exhibiting *cis*-regulatory divergence with scans for selection identified several *cis*-regulatory genes under positive selection, consistent with a role for these loci in local adaptation. The combination of allele-specific expression with genomic scans is a fruitful approach to identify the regulatory architecture of adaptive evolution in natural populations.

Comparisons between New York and Brazil house mice provide insights into the evolution of gene regulation over very short evolutionary timescales. Although New York and Brazil house mice colonized the Americas only within the last ~500 years, we find evidence for pervasive regulatory divergence. Moreover, the regulatory control underlying this intraspecific divergence is overwhelmingly due to *cis* variants either alone or together with one or more *trans* variants. The predominance of *cis*-regulatory divergence is in agreement with previous interspecific studies in house mice [@Goncalves2012; @Crowley2015; @Mack2016], and has been observed in intraspecific comparisons of other species [@Lemmon2014; @Genissel2008; @Verta2019; @Osada2017; @Lovell2016; @Gould2018; @Fear2016]. Although some evidence suggests *trans*-effects play larger roles within rather than between species [@Coolon2014; @Hart2018; @Jacobs2022; @Lemos2008; @Emerson2010], it is likely that certain evolutionary contexts, timescales, and selection pressures may favor either *cis*- or *trans*-acting mechanisms [@Signor2018; @Buchberger2019]. Regardless, our study indicates that strong intraspecific *cis*-regulatory divergence between populations can accrue on extremely short timescales.

Despite the plastic response of gene expression in both New York and Brazil house mice, *cis*-regulatory divergence was relatively robust to environmental temperature. In contrast, changes in the environment preferentially affected *trans*-regulation profiles, suggesting that *trans*-effects play a more pronounced role in gene expression plasticity. Greater sensitivity of *trans*-effects to the environment is in strong agreement with previous studies [@Smith2008; @Tirosh2009; @Li2006; @Chen2015; @Cubillos2014; @Verta2019; @Li2017] and may be due to the role *trans*-acting factors play in signaling pathways that become activated in response to environmental change [@Ehrenreich2016]. Indeed, we found that the effect sizes of *trans* were greater than those of *cis* across environments, indicating that much of expression plasticity we observed is governed by changes in *trans*. Moreover, the pronounced expression plasticity we observe in Brazil house mice largely goes in the same direction as evolved divergence (i.e., adaptive plasticity)[@Ghalambor2015; @Fischer2016]. Previous studies in house mice have implicated the role of adaptive gene expression plasticity in local adaptation [@Bittner2021], suggesting that plasticity in general may have aided in the colonization of new environments.

Although ASE was generally observed at both temperatures for a given gene, a subset of genes showed temperature-dependent *cis*-effects. These loci are of particular interest since these constitute *plasticity*-eQTL and harbor mutations that directly affect plasticity of gene expression. Genetic assimilation refers to the conversion of a plastic response to a fixed response [@Waddington1942; @Waddington1952; @Waddington1953; @Van_der_Burg2020]. If the ancestral allele at a *plasticity*-eQTL encodes a plastic response and the derived allele encodes a fixed response, then the *plasticity*-eQTL represents a case of genetic assimilation. For example, selection in a cold, temperate environment may have led to the reduced plasticity exhibited in New York mice. A similar mechanism was recently proposed to underlie rapid divergence in threespine stickleback [@Verta2019]. *Cis*-regulatory variants could rapidly canalize expression through the loss or gain of specific binding sites for conditionally expressed transcription factors, thereby decoupling a gene’s expression from the environment [@Ehrenreich2016]. Many of the *cis* x environment candidates illustrate potential regulatory mechanisms underlying genetic assimilation as many of them exhibit reduced plasticity in New York mice (Figure S11). For example, *scd1* plays an important role in basal and cold-induced thermogenesis [@Ntambi2002; @Lee2004] and New York mice show higher and constitutive average expression of *scd1* in BAT compared to Brazil mice (Figure S11). Further study of these genes may help us understand the relationship between adaptive plasticity and genetic adaptation to novel environments.

Finally, we discovered significant overlap between genes exhibiting *cis*-regulatory divergence and genomic SNPs that show evidence for positive selection in wild mice, suggesting that selection has acted mainly on regulatory variants associated with local adaptation. This overlapping gene set is enriched for mutant phenotypes related to body size and metabolism in New York mice and are consistent with previous studies showing selection on genes with *cis*-eQTLs related to body size in North American mice [@Mack2018]. Together, our results highlight how natural selection on *cis*-regulatory divergence is a likely contributor to rapid climatic adaptation in house mice.

Materials and Methods {.unnumbered}
========================================

### Animals and Evolved Phenotypic Differences {#animals-and-evolved-phenotypic-differences .unnumbered}

To characterize evolved phenotypic differences between New York and Brazil house mice, we used two wild-derived inbred lines of house mice: SARA (New York) and MANA (Brazil). The establishment of these lines has been described previously [@Ballinger2022]. Mice from each line were housed in a standard laboratory environment at 21$^{\circ}$C with a 12L:12D cycle. Roughly equal numbers of males and females were produced for each within-line comparison (*n* = 32 per line; File S1). We took standard museum measurements on all mice and removed and prepared dried skins. Thermal conductance of pelage (referred to as pelage conductance (W^-1^m^-2^C^-1^)) was measured on dry skins following the protocol of Riddell et al. 2021 (see SI Methods) [@Riddell2021]. Tail length and ear length were corrected for body mass for each individual. Effects of line and sex for each phenotype were modeled using ANOVA. All statistical analyses were performed using packages available in R (v.4.1.1).

### Experimental Design and Tissue Collection {#experimental-design-and-tissue-collection .unnumbered}

To investigate the gene regulatory mechanisms underlying local adaptation in house mice, we generated F1 hybrids by crossing a SARA female with a MANA male. All experimental animals were born at room temperature (21$^{\circ}$C) and were provided water and commercial rodent chow *ad libitum*. We weaned and singly housed SARA, MANA, and F1 hybrids at ~3 weeks of age. We split 3.5-week-old full-sibs and F1 hybrids into size-matched experimental groups across cold (5$^{\circ}$C) and warm (21$^{\circ}$C) treatments. Mice were kept in their respective experimental environment until ~12 weeks of age, at which point individuals were euthanized via cervical dislocation. We took standard museum measurements and then rapidly dissected and preserved liver and brown adipose tissue in RNAlater at 4$^{\circ}$C overnight and moved to -80$^{\circ}$C until RNA extraction. We prepared standard museum skeletons and accessioned them in UC Berkeley’s Museum of Vertebrate Zoology (catalog numbers are given in File S1). All experimental procedures were in accordance with the UC Berkeley Institutional Animal Care and Use Committee (AUP-2017-08-10248).

### RNA Extraction, Library Preparation, and Sequencing {#rna-extraction-library-preparation-and-sequencing .unnumbered}

We extracted total RNA from liver and BAT from each sample (*n* = ~6 per genotype/sex/treatment/tissue) using the RNeasy PowerLyzer Kit (QIAGEN). We generated Illumina cDNA libraries from 1 $\mu$g of purified RNA using KAPA Stranded mRNA-Seq Kit (Illumina), and uniquely indexed libraries using unique dual indexes (Illumina). Libraries were pooled in equal molar concentration and sequenced on one lane each of 150 bp paired-end NovaSeq S1 and NovaSeq S4 at the Vincent J. Coates Genomics Sequencing Center at UC Berkeley. We filtered raw reads below a Phred quality score of 15 and trimmed adapter sequences using fastp [@Chen2018].

### Parental Gene Expression Analyses {#parental-gene-expression-analyses .unnumbered}

After cleaning and trimming parental sequences of MANA and SARA, we mapped reads to the *Mus musculus* reference genome (GRCm38/mm10) using STAR [@Dobin2013]. We counted reads overlapping exons using HTSeq [@Anders2015] based on the Ensembl GRCm38.98 annotation. We imported raw count data into R (v.4.1.1) and transformed expression values using variance stabilizing transformation [@Love2014] to assess transcriptome-wide expression patterns via PCA. Next, we removed genes with fewer than an average of 10 reads per individual within each tissue, retaining ~14K expressed genes per tissue for downstream analyses. We then used DESeq2 [@Love2014] on raw, filtered reads to quantify expression patterns by fitting a generalized linear model following a negative binomial distribution. We computed differential expression between lines with the model population + environment + population*environment to determine the effects of genotype, environment, and genotype-by-environment on expression patterns for each tissue and sex, separately. We also identified genotype-by-sex interactions using a similar model in DESeq2 (see SI Methods and Results).

To determine if gene expression plasticity is correlated with gene expression divergence, we compared genes with significant plasticity to genes with significant expression divergence within each tissue and sex, separately. We used Spearman’s rank correlation coefficients to assess overall directionality and significance of gene expression. To account for potential statistical artifacts [@Mallard2018], we compared the observed correlations to a permuted distribution (10,000 permutations). Lastly, we used a Benjamini-Hochberg multiple test correction [@Benjamini1995] on all resulting \textit{P}-values and considered genes with FDR < 0.05 to be significantly differentially expressed.

### Identifying Variants between Parental Lines {#identifying-variants-between-parental-lines .unnumbered}

To identify differences between lines for allele-specific read assignment, we performed SNP calling on whole genome sequence data from one female each of MANA and SARA. We mapped genomic reads with Bowtie2 [@Langmead2012] to the mm10 reference genome (setting: --very-sensitive) obtained from Ensembl. We marked duplicates with the Picard tool MarkDuplicates and then we used the GATK tools HaplotypeCaller and GenotypeGVCFs for joint genotyping across genomic samples. We filtered for low quality SNP calls with VariantFiltration (QD < 2.0; QUAL < 30.0; FS > 200; ReadPosRankSum < -20.0). To reduce the influence of genotyping error on allele-specific expression, we mapped RNA-seq reads from all individuals and then counted allele-specific reads aligned to each site we genotyped with the GATK tool ASEReadCounter. We excluded sites for which we did not have coverage of at least 5 reads from each population-specific allele. These SNPs were then used for identifying allele-specific reads.

### Mapping Allele-Specific Reads {#mapping-allele-specific-reads .unnumbered}

For allele-specific expression analyses, we mapped reads from hybrid individuals to the mouse reference genome (GRCm38/mm10) using STAR. We used WASP [@Van_de_Geijn2015] to reduce the potential for reference mapping bias. We retained reads that overlapped a population-specific variant and that passed WASP filtering for our allele-specific expression analysis. We separated reads overlapping informative variants into allele-specific pools (NY, BZ) based on genotype for quantification. We used HTSeq to count the number of reads associated with each gene per population based on the overlap of reads and annotated exonic regions based on the Ensembl GRCm38.98 annotation. We examined per site allelic reads with ASEReadCounter to quantify allele-specific mapping over individual sites. Proportions of reads overlapping the references vs. alternative allele (REF allele / (ALT allele + REF allele)) showed a median 0.5 across samples (Figure S12), indicating no evidence for reference mapping bias.

### Identifying *Cis*- and *Trans*-Regulatory Divergence {#identifying-cis-and-trans-regulatory-divergene .unnumbered}

Parental (F0) and F1 expression data was used to characterize *cis* and *trans* effects. To categorize regulatory divergence at each gene, we inferred differential expression by analyzing raw counts using DESeq2. To identify genes with evidence of allele-specific expression in hybrid individuals, we took reads that mapped preferentially to either New York or Brazil alleles and fit these to a model with allele (NY vs. BZ), sample (individual), and tissue (BAT, liver) for hybrid male samples in DESeq2 (Wald-test). As read counts come from the same sequencing library, library size factor normalization was disabled in DESeq2 by setting SizeFactors = 1 for measures of allele-specific expression. We used males to assign regulatory categories to maximize power due to a larger number of hybrid samples sequenced (6 replicates of males vs. 4 replicates of females). Differential expression between alleles in the F1 is evidence for *cis*-regulatory divergence, where differential expression in the F0 generation is not recapitulated between alleles in the F1 is evidence for *trans* divergence. The *trans* component (T) was assessed through a Fisher’s Exact Test on reads mapping to each parental allele in the hybrid vs. parental read counts, summed over all replicates [@McManus2010; @Coolon2014]. Reads were randomly down-sampled to account for library size differences between parental and F1 replicates [@Coolon2018; @Mack2017]. *P*-values for each test were corrected for FDR with the Benjamini-Hochberg method. Genes were sorted into categories based on hard FDR thresholds (FDR < 0.05) [@McManus2010; @Coolon2014], as described below. We analyzed temperature treatments (warm and cold) separately for regulatory assignment and then compared as described below: \
\indent \textit{Conserved}: no significant difference between lines (F0), no significant difference between alleles (F1), no significant \textit{T}. \
\indent \textit{Cis} only: significant difference between lines (F0), significant difference between alleles (F1), no significant \textit{T}. \
\indent \textit{Trans} only: significant difference between lines (F0), no significant difference between alleles (F1), significant \textit{T}. \
\indent _Cis_ <span>&#38;</span> _Trans_ designations: significant differences between alleles (F1) and significant \textit{T}. This category was further subdivided into _cis_ + _trans_ (reinforcing), _cis_ + _trans_ (opposing), \textit{compensatory}, and _cis_ x _trans_, as previously described [@Goncalves2012; @Mack2016]. \
\indent _Ambiguous_: all other patterns.

We identified *cis* x temperature interactions using DESeq2 under a model specifying temperature (cold vs. warm) and allele (BZ vs. NY). To identify *trans* x temperature interactions, we fit a model that included parental and hybrid read counts for temperature (cold vs. warm), allele/genotype (BZ vs. NY), and generation (F1 vs. F0) and interactions. Similar models were also used to identify sex-specific regulatory patterns in DESeq2 (see SI Methods and Results).

### Genetic PCA of *M.m. domesticus* populations {#genetic-PCA-of-m.m-domesticus-populations .unnumbered}

We used SNPRelate [@Zheng2012] to perform PCA and IBS hierarchical clustering of population genetic data. Genomic data from 3 Eurasian populations of *M. m. domesticus* (Germany [Cologne-Bonn], France, and Iran) and *M. m. musculus* and *M. m. castaneus* subspecies were downloaded from http://wwwuser.gwdg.de/~evolbio/evolgen/wildmouse/ [@Harr2016]. For PCA, biallelic variants genotyped across all these individuals were extracted and pruned for linkage disequilibrium in SNPRelate (thresholds=0.2) resulting in 22,126 variant sites for PCA and IBS clustering for *M. m. domesticus* comparisons and 25,467 variants for global *Mus* comparisons (Figures 4A, S9). Altering the pruning threshold to 0.5 did not result in any change in population clustering.

### Autosomal Scans for Selection {#autosomal-scans-for-selection .unnumbered}

To identify regions with evidence for selection in the Americas, we scanned the exomes of our North and South American focal populations for selection by using a modification of the population branch statistic (PBS) which summarizes a three-way comparison of allele frequencies between a focal group, a closely related population, and an outgroup comparison (*PBSn1*) [@Yi2010; @Crawford2017]: $$ PBSn1 = {PBS_1 \over 1 + PBS_1 + PBS_2 + PBS_3}  $$

Here, *PBS1* indicates PBS calculated as either Manaus or NH/VT as the focal population, and *PBS2* and *PBS3* indicate PBS calculated for Eurasians populations as the focal populations (France or Germany and Iran, respectively). To maximize the number of sites that could be compared, American populations are not directly compared in the branch test due to the reduced representation of exome data and high per site Fst values between the two populations (Figure S13). Instead, NH/VT and MAN were each compared to two Eurasian populations [((MAN), France) Iran) and ((NH/VT) Germany) Iran)], selected based on population clustering (Figure S9). We restricted our SNP set to biallelic variants across the 3 populations being compared and required that at least six individuals in the focal branch be genotyped. We note that the NH/VT sample used in the PBS test is geographically close to the origin of the SARA line.

We used VCFtools [@Danecek2011] to calculate Weir and Cockerham Fst at each variant position. These values were used to calculate *PBSn1* for non-overlapping blocks of 5 SNPs. We consider blocks in the top 1% of *PBSn1* scores outliers and do not attempt to assign *P*-values to each SNP-block [@Malaspinas2016]. Outliers were >3 standard deviations above the mean windowed value of SNP-blocks in each comparison (MAN focal, median=0.045; NH/VT focal median = 0.064). We identified windows overlapping genes based on Ensembl gene coordinates (mm10) and the BEDTools “intersect” tool [@Quinlan2010]. As allele-specific expression in F1s is consistent with local independent genetic changes influencing gene expression, we focused on genes with evidence for *cis*-regulatory divergence (i.e., differences in expression between parental alleles in the F1) for overlap with outlier loci. To ask whether allele-specific expression was associated with elevated *PBSn1* scores, we used a generalized linear model incorporating gene category (ASE or no ASE) and SNP density per kb as factors to *PBSn1* scores. SNP density was calculated by dividing the number of informative sites between NY and BZ for allele-specific expression per gene by transcript length.

### Enrichment Analyses {#enrichment-analyses .unnumbered}

We performed all GO and pathway enrichment analyses with PANTHER [@Mi2013; @Thomas2003]. Phenotype enrichment analyses were performed with ModPhea [@Weng2017]. We annotated genes to specific phenotypes based on Mouse Genome Informatics phenotype annotations (http://www.informatics.jax.org/).

### Data Availability {#data-availability .unnumbered}

Scripts are available on GitHub(https://github.com/malballinger/BallingerMack_NYBZase_2022). All sequence data generated in this study have been deposited to the National Center for Biotechnology Information Sequence Read Archive under accession BioProject ID PRJNAXXX. All other data are included in the article and/or SI Appendix.

<!-- Leave these lines as they are at the end of your .Rmd file to ensure placement of methods & acknowledgements sections before the references-->
\showmatmethods
\showacknow
\pnasbreak

References{#references .unnumbered}
========================================